Core/Core/Inbox/AttachmentPicker/Model/AttachmentPickerInteractor/AttachmentPickerInteractorLive.swift

```
@@ -68,10 +106,87 @@ class AttachmentPickerInteractorLive: AttachmentPickerInteractor {
+            fileStore.refresh()
+        }
+    }
+
+    func deleteFile(file: File) -> AnyPublisher<Void, Never> {
         uploadManager.viewContext.delete(file)
         fileStore.refresh()
+
+        if let fileId = file.id {
+            return ReactiveStore(useCase: DeleteFile(fileID: fileId))
+                .getEntities()
+                .mapToVoid()
+                .replaceError(with: ())
+                .eraseToAnyPublisher()
+        } else {
+            return Just(()).eraseToAnyPublisher()
+        }
+    }
+
+    private func duplicateFileToUploadFolder(file: File) {
+        if let fileId = file.id {
+            getOnlineFileURL(fileId: fileId)
+                .map { [weak self] url in
+                    if let self, let url {
+                        _ = try? self.uploadManager.add(url: url, batchID: self.batchId)
+                        let newValues = alreadySelectedFiles.value.filter { $0 != file }
+                        alreadySelectedFiles.send(newValues)
+                    }
+                }
+                .sink()
+                .store(in: &subscriptions)
+        }
+    }
+
+    private func getOnlineFileURL(fileId: String) -> AnyPublisher<URL?, Error> {
+        ReactiveStore(useCase: GetFile(context: .currentUser, fileID: fileId))
+            .getEntities()
+            .map { files in
+                return files.first?.url
+            }
+            .flatMap { [weak self] url in
+                if let url,
+                   let self,
+                   let session = self.env.currentSession,
+                   let request = try? url.urlRequest(relativeTo: session.baseURL, accessToken: session.accessToken, actAsUserID: session.actAsUserID) {
+                    return self.publisherProvider.getPublisher(for: request)
+                } else {
+                    return Fail(error: NSError.instructureError(String(localized: "Failed to duplicate file", bundle: .core))).eraseToAnyPublisher()
```

Do we need to localize it? It seems to me we silence this error anyway.